skill(next-cache-components-adoption): clarify prereqs (#95082)
vercel/next.js: docs/01-app/02-guides/migrating-to-cache-components.mdx

> **Good to know:** If you were using `experimental.dynamicIO` or `experimental.useCache`, `cacheComponents` replaces them. See the [version 16 upgrade guide](/docs/app/guides/upgrading/version-16#experimentaldynamicio-and-experimentalusecache).

Before enabling the flag, remove every `export const dynamic` from your route segments. `cacheComponents: true` rejects pages that still export this config.

> Other legacy APIs (`revalidate`, `fetchCache`, `unstable_cache`, and the `fetch` cache options) keep working as a separate caching layer after the flag is on. The sections below cover how to translate each one. Migrate them as the validation insights and errors point at each one.

Author: Verified against the framework source — only `export const dynamic` is rejected by `cacheComponents: true` (see `packages/next/src/server/app-render/create-component-tree.tsx`). `export const revalidate` and `export const fetchCache` keep working alongside `cacheComponents`; the migration guide sections below show how to translate them to `cacheLife` and `use cache` after the flag is on. Not changing the guidance.

On-demand invalidation still works by tagging cached data and expiring it after an event. Tag data with [`cacheTag`](/docs/app/api-reference/functions/cacheTag) inside a `use cache` function instead of the `fetch` `next.tags` option, then choose the invalidation API by the behavior you want:

- [`updateTag`](/docs/app/api-reference/functions/updateTag): for mutations whose result the user must see immediately (read-your-own-writes). Called from a Server Action, it expires the tag so the next request waits for fresh data instead of serving stale content.

> - [`revalidateTag`](/docs/app/api-reference/functions/revalidateTag): for stale-while-revalidate. Pass a cache profile like `'max'` as the second argument to serve cached data while it refreshes in the background. The single-argument form `revalidateTag(tag)` is deprecated under Cache Components and may be removed in a future version. Works in Server Actions and Route Handlers.

Reviewer: > The single-argument form `revalidateTag(tag)` is deprecated under Cache Components and may be removed in a future version

This is deprecated in v16, not only with CC

Reviewer: ```suggestion
- [`revalidateTag`](/docs/app/api-reference/functions/revalidateTag): for stale-while-revalidate. A cache profile is **required** as second argument, for example use `'max'` to serve cached data while it refreshes in the background. Works in Server Actions and Route Handlers.
```
